### ISSUE TITLE
Show full name in user mentions autocompletion suggestion in addition to username and allow search by it

### ISSUE THREAD

**Issue by Gim6626** [taigaio/taiga-front#1859]
I'd like to request feature of showing full names in autocompletion user mentions suggestions in addition to user names and allowing search by it.

Currently if I for example write a comment and want to mention a user, I could press `@`, start typing user name and get drop down list which allows to choose existing users whose usernames begin with text that I type. But it is not so simple and intuitive to find user name. Full names are much more common in user interface. I see that also you could start typing full name of user and autocompletion popup will appear also. But if I for example know that name of user is "John" and don't know his username, there could be many Johns in big project and in autocompletion popup I'll see something like:

- @johnallmighty1232131
- @johncoder321321
- ...

And how should I understand who is needed John here?

I think it will be really useful if I could start typing `John` and get autocompletion suggestions both with usernames and fullnames, like:
- @johnallmighty1232131 - John Smith
- @johncoder321321 - John Johnson
- ...

Such suggestion list is much more useful!

In same way user mentions autocompletion are implemented for example in GitHub, see screenshot:

![Screenshot_20200113_102751](https://user-images.githubusercontent.com/169866/72234545-5fdd0e80-35ef-11ea-85a4-ec6f33621f05.png)

**Comment by theriverman** [comment]
This is a very good proposal indeed and I'd like to extend it with the way mentions get rendered.
That is, showing the user's full name in the \@mention filter menu is one thing, but rendering the user's `full_display_name` instead of their `username` in the HTML/Markdown text would be a huge improvement too.
The information is available, it's just needs to be adjusted in the frontend.

**Comment by theriverman** [comment]
I've forked the frontend and began playing around with the impacted modules. I already have a working solution for the initial request mentioned in this issue, and now looking for a solution to reverse-render the user's `fullname` property in the wysiwyg boxes.

@migonzalvar or somebody else: could you put this item on the roadmap mentioning me to avoid multiple people work in parallel on the same issue, please?

**Comment by migonzalvar** [comment]
@theriverman I can assure you that taiga team developers will contact you before starting to work on this feature, don't worry about it

**Comment by theriverman** [comment]
Thanks, @migonzalvar 
I'm progressing a bit slowly, not much free time these days, but I'll be ready with the first draft in a couple of days.

**Comment by juanfran** [comment]
This will be include in [taiga-front-next](https://github.com/taigaio/taiga-front-next)

**Comment by NatachaMenjibar** [comment]
We have been doing some housekeeping work prior to Taiga6 upcoming release and we are reviewing old issues that we didn't correctly attend back then. Since Taiga6 brings a significant UX/UI rework, we encourage you to wait and see if Taiga6 satisfies this need and open a new issue if that's not the case. This time we will make sure to review it when it comes through. 

Thank you very much on behalf of the Taiga Team!
